spark8t/domain.py

```diff
@@ -176,30 +176,37 @@ def __init__(self, environ: Dict = dict(os.environ)):
         self.environ = environ if environ is not None else {}
 
     @property
-    def spark_folder(self) -> str:
-        """Return the SNAP folder"""
-        return self.environ.get("SNAP", self.environ.get("SPARK_HOME"))
+    def spark_home(self):
+        return self.environ["SPARK_HOME"]
 
     @property
-    def static_conf_file(self) -> str:
-        """Return static config properties file packaged with the client snap."""
-        return f"{self.spark_folder}/conf/spark-defaults.conf"
+    def spark_confs(self):
+        return self.environ.get("SPARK_CONFS", os.path.join(self.spark_home, "conf"))
+
+    @property
+    def spark_user_data(self):
+        return self.environ["SPARK_USER_DATA"]
+
+    @property
+    def kubectl_cmd(self) -> str:
+        """Return default kubectl command."""
+        return self.environ.get("SPARK_KUBECTL", "kubectl")
+
+    @property
+    def kube_config(self) -> str:
+        """Return default kubeconfig to use if not explicitly provided."""
+        return self.environ["KUBECONFIG"]
```

Comment:
Previously we've allowed for an automatic discovery of a default kubeconfig file.
If I'm getting it right, this feature is now removed. Instead we strictly insist on an external definition of the kube config location, with no in-built fallback default. 
Why do we prefer to go this way?